src/backend/xilinx/axi_address_space.rs

```diff
@@ -294,8 +306,12 @@ impl AddressSpace {
         }
 
         // port writes to internal register logic
+        // reads only
+        // XXX(nathanielnrn) Why does this need to be seperate from the above write logic?
+        // Seems basically the same to me?
```

Comment:
Indeed; maybe this could be factored out into a function? Or maybe it would be too annoying because of the slight differences in flag values…

Reply:
Gave this a shot, indeed a bit annoying because of scope issues, the helper function has a bunch of arguments and imo is not much cleaner than how it is currently. I think refactoring this could be a small issue for later? Willing to work on this more though.

Reply:
Yeah, totally, makes sense to me! I can imagine how this would imply a broader refactoring that wouldn't be trivial.